src/deferred/exceptionHook.js

```diff
@@ -0,0 +1,21 @@
+define( [
+	"../core",
+	"../deferred"
+], function( jQuery ) {
+
+// These usually indicate a programmer mistake during development,
+// warn about them ASAP rather than swallowing them by default.
+var rerrorNames = /^(Eval|Internal|Range|Reference|Syntax|Type|URI)Error$/;
+
+jQuery.Deferred.exceptionHook = function( error, stack ) {
+
+	// Support: IE9
+	// Console exists when dev tools are open, which can happen at any time
+	var warn = window.console && window.console.warn;
+
+	if ( warn && error && rerrorNames.test( error.name ) ) {
+		warn.call( window.console, "jQuery.Deferred exception: " + error.message, stack );
```

Comment:
Not that this matters in the slightest, but seeing as how you're accessing the global immediately above this as feature detection, is there any reason to `.call` rather than just `window.console.warn(...)`?  Might even save some bytes ;)